Code/Core/CoreTest/Tests/TestSharedMemory.cpp

```diff
@@ -0,0 +1,96 @@
+// TestSharedMemory.cpp
+//------------------------------------------------------------------------------
+
+// Includes
+//------------------------------------------------------------------------------
+#include "TestFramework/UnitTest.h"
+
+// Core
+#include <Core/Env/Assert.h>
+#include <Core/Process/SharedMemory.h>
+#include <Core/Process/SystemMutex.h>
+#include <Core/Process/Thread.h>
+
+
+#if defined(__LINUX__) || defined(__APPLE__)
+	#include <sys/types.h>
+	#include <wait.h>
+	#include <unistd.h>
+#endif
+
+// TestSharedMemory
+//------------------------------------------------------------------------------
+class TestSharedMemory : public UnitTest
+{
+private:
+	DECLARE_TESTS
+
+	void CreateAccessDestroy() const;
+};
+
+// Register Tests
+//------------------------------------------------------------------------------
+REGISTER_TESTS_BEGIN( TestSharedMemory )
+	REGISTER_TEST( CreateAccessDestroy )
+REGISTER_TESTS_END
+
+// CreateAccessDestroy
+//------------------------------------------------------------------------------
+void TestSharedMemory::CreateAccessDestroy() const
+{
+
+#if defined(__WINDOWS__)
+	// TODO:WINDOWS Test SharedMemory (without fork, so).
+#elif defined(__LINUX__) || defined(__APPLE__)
+	SystemMutex mutex("fbuild_shared_memory_test");
+	bool locked = mutex.TryLock();
+
+	(void)locked;
+	ASSERT(locked);
```

Comment:
Use TEST_ASSERT. This is not stripped in debug and will remove the need to use (void)x to avoid unused variable wanings.